File: scratch/replace_print.dart
import 'package:tax_flow_pro/utils/logger.dart'; // this is just a dummy to avoid errors, actually this will run without pub get if we use simple dart.
import 'dart:io';

void main() {
  var dir = Directory('c:/Users/Hp/Documents/Sviluppo/contabile/contabile_app/lib');
  
  for (var entity in dir.listSync(recursive: true)) {
    if (entity is File && entity.path.endsWith('.dart')) {
      var content = entity.readAsStringSync();
      if (content.contains('print(')) {
        var lines = content.split('\n');
        var newLines = <String>[];
        var madeChanges = false;
        
        for (var line in lines) {
          if (line.contains('print(')) {
            if (line.toLowerCase().contains('errore')) {
              newLines.add(line.replaceAll('print(', 'appLogger.e('));
            } else {
              newLines.add(line.replaceAll('print(', 'appLogger.i('));
            }
            madeChanges = true;
          } else {
            newLines.add(line);
          }
        }
        
        if (madeChanges) {
          var newContent = newLines.join('\n');
          if (!newContent.contains('logger.dart')) {
            var importStmt = "import 'package:tax_flow_pro/utils/logger.dart';\n";
            if (newContent.contains('import ')) {
              newContent = newContent.replaceFirst('import ', importStmt + 'import ');
            } else {
              newContent = importStmt + newContent;
            }
          }
          entity.writeAsStringSync(newContent);
        }
      }
    }
  }
  print('Done');
}

File: scratch/replace_print.py
import os
import re

lib_dir = "c:/Users/Hp/Documents/Sviluppo/contabile/contabile_app/lib"

for root, _, files in os.walk(lib_dir):
    for file in files:
        if file.endswith(".dart"):
            path = os.path.join(root, file)
            with open(path, "r", encoding="utf-8") as f:
                content = f.read()
            
            if "print(" in content:
                # Replace print with appLogger.i (or e if it says Errore)
                lines = content.split("\n")
                new_lines = []
                made_changes = False
                for line in lines:
                    if "print(" in line:
                        if "errore" in line.lower():
                            new_lines.append(line.replace("print(", "appLogger.e("))
                        else:
                            new_lines.append(line.replace("print(", "appLogger.i("))
                        made_changes = True
                    else:
                        new_lines.append(line)
                
                if made_changes:
                    # check if logger is imported
                    new_content = "\n".join(new_lines)
                    if "logger.dart" not in new_content:
                        import_stmt = "import 'package:tax_flow_pro/utils/logger.dart';\n"
                        # insert after the first import, or at top
                        if "import " in new_content:
                            new_content = new_content.replace("import ", import_stmt + "import ", 1)
                        else:
                            new_content = import_stmt + new_content
                            
                    with open(path, "w", encoding="utf-8") as f:
                        f.write(new_content)
                
print("Done")
